I1 seir model

FarmHJ opened the pull request:
Addresses #1. Creates a class for SEIDR model. Unit tests, documentations and vignettes are included.

solveigvandervegt committed:
Add SEIR model as a S4 object

Amended from the S4 object created by Ioana and Oscar for the age-structured model. Documention added for this class and the packages it depends on have been to the DESCRIPTION file.

FarmHJ committed:
Try calling SEIR class with test

FarmHJ committed:
Create tests for SEIR_model and update documentation

FarmHJ committed:
Check all tests pass and no style issue

solveigvandervegt committed:
Added plotting back

Added plotting as an option to the simulate function under the flag is_plot. Default is TRUE

solveigvandervegt committed:
Create vignette for SEIR_model

Added a folder for the vignettes and made a start on the vignette explaining the SEIR_model class and how to use it. Doesn't knit at the moment unless you comment out library(comomodel).

solveigvandervegt committed:
Update vignette with code

Update the vignette to include a section on how to call the model. Not working yet cause for some reason it can create an SEIR_model but then it cannot find the method set_parameters

FarmHJ committed:
Add export function and edit minor mistakes on vignette

FarmHJ committed:
Change population name from 'infected' to 'infectious'

FarmHJ committed:
Separate parameters, setter getter method, include incidence

FarmHJ committed:
Edit tests according to changes in SEIR class

FarmHJ committed:
Create SEIDR model

solveigvandervegt committed:
Updates to handling of C0 and D0

Removed setter function for initial condition on cases and deaths. We now compute/set these in the initial conditions function but the user shouldn't touch these. We only count deaths from t=0. Changed names of slots and associated functions accordingly

Changed name of parameter a to parameter m, because I believe mu is the conventional name for the death rate.

Changed name of simulate_SEIR to simulate_SEIDR to stick with naming convention across models.

Do we need to reformat the output?

solveigvandervegt committed:
Edited vignetted

Added death rates and new variables C and D.

Still trying to get the code to work with the new functions.

solveigvandervegt committed:
Fighting setter methods

Changed the argument names in the setter function. Apparantly the second argument must be named value, and it must be one argument. Rewritten these methods to work for this.

For some reason the test code at the bottom os SEIDR.R works fine but trying to run the exact same lines of code in the vignette still doesn't. No clue why, will try again later.

ben18785 committed:
Removed duplicate generics and changed signature of those in SEIR_model

solveigvandervegt committed:
Update vignette for SEIDR class

Fixed some typos and phrasing.

Added a plot at the end. Will need to edit this when we have general plotting function.

FarmHJ committed:
Update tests according to changes in class

FarmHJ committed:
Output incidences instead of components and lintr check

FarmHJ committed:
Update documentation

FarmHJ committed:
Create new plot for incidences and deaths

solveigvandervegt committed:
Fixed documentation problems

Generics and methods in S4 classes need to be documented separately or devtools::check() throws warnings. That is done now and there is no more warning but the documentation itself might be a bit messy at the moment

solveigvandervegt committed:
Cleaned up documentation

Deleted double documentation of methods and generics

solveigvandervegt committed:
Deleted @describeIn

Was trying to have the methods documented in the same file as the class but the @describeIn tag doesn't work in combination with the @rdname tag. SInce the documentation now works and no more warnings are thrown by devtools::check() we might just have to accept that the methods are documented in separate files

FarmHJ committed:
Change output format for consistency across models

FarmHJ committed:
Update documentation to include required package

FarmHJ committed:
Edit tests to take in long format output

FarmHJ committed:
Update vignette according to changed format of simulation output

FarmHJ committed:
Merge branch 'main' into i1-seir-model

FarmHJ committed:
Add markdown requirement for vignette builder

FarmHJ committed:
Follow suggestion of error message to build vignette

FarmHJ committed:
Try to remore unnecessary packages

FarmHJ committed:
Include rmarkdown package to Suggests packages

solveigvandervegt committed:
Edited documentation

Changed from @rdname tag to @describeIn tag to add documentation for the methods to the same file as the documentation of the class.

FarmHJ committed:
Adjust coding style (lintr)

solveigvandervegt committed:
Changed SEIDR to SEIRD

Putting the D last in the name is more consistent with literature.

oscardm20994 commented on R/SEIRD.R:
```diff
@@ -0,0 +1,347 @@
+    names(output) <- c("time", "compartment", "value")
+
+    # Added for consistency of output format across models
+    output$age_group <- rep("0-150", length(output$time))
```
We should ensure the long format of the output is consistent with the SEIRAge model, this looks good but we've done it slightly differently. both would probably work well.

solveigvandervegt replied:
Thanks Oscar, I will change ours to match yours!

oscardm20994 reviewed (commented):
This is looking really good, @solveigvandervegt and @FarmHJ. Just need to make sure there is consistency across this and the age model in outputs etc. Other than that looks in really good shape.

solveigvandervegt committed:
Edit df output format

Edited order of columns in output dataframe from simulate function to match exactly the formatting in the age-structured model.

solveigvandervegt committed:
Final edits

changed name of simulate function to run()
changed output to be two dataframes, one with SEIR data and one with incidences and deaths
Edited tests and vignette to match changes made
Changed class definition such that a new object of the class can now be created with newmodel <- SEIRD()

solveigvandervegt committed:
fixed typo in vignette

solveigvandervegt commented:
@ben18785 I think I made all the changes that we talked about. Let me know if I missed anything!

solveigvandervegt committed:
Removed testing lines

ben18785 committed:
A number of changes to SEIRD class and vignette

- removed initial cases and deaths setters / getters
- made setters take named arguments for initial_conditions and transmission_parameters
- changed intro to vignette a bit to make it less focussed on the code; more on the models themselves

ben18785 committed:
Added R0 functionality

ben18785 committed:
Added test of R0 and also informative error messages for SEIRD

solveigvandervegt committed:
Implement suggestions by lint check

solveigvandervegt committed:
Edited vignette for clarity

solveigvandervegt committed:
Edits to vignette

I think we need to be careful with the wording as models with R_0>1 also eventually have the epidemic die out due to low S. I think this is mainly about the initial growth and I think that needs to be made very clear.

solveigvandervegt committed:
Edits to vignette

solveigvandervegt committed:
Editing plotting of D and cases

edit to plot fraction of population per day and barplot

solveigvandervegt committed:
Vignette plots edits

Edited vignette so all incidences and deaths are now plotted as bar graphs and with units of per day with respect to time.

ben18785 reviewed (approved):
Great work guys! I'm happy to merge now.
